portfolio/src/main/webapp/script.js

```diff
@@ -20,28 +20,116 @@ import { photosData } from "./photos-data.js";
  * @return {html component}
  */
 function createPhoto(photo) {
-  const component = `<div class="col-4"> \
-          <figure class=\"figure\"> \
-              <img src=${photo.src} class=\"figure-img img-fluid rounded\" alt=${photo.location}> \
-              <div class=\"row figure-caption-container\"> \
-                  <figcaption class=\"figure-caption photo-location\">${photo.location}</figcaption> \
-                  <span class=\"dot\"></span> \
-                  <figcaption class=\"figure-caption photo-date\">${photo.date}</figcaption> \
-              </div> \
-          </figure> \
-      </div>`;
-
-  return component;
+  const component = [
+    "<div id=" + photo.epoch + ' class="col-4">',
+    '<figure class="figure">',
+    "<img src=" +
+      photo.src +
+      ' class="figure-img img-fluid rounded" alt=' +
+      photo.location +
+      ">",
+    '<div class="row figure-caption-container">',
+    '<figcaption class="figure-caption photo-location">' +
+      photo.location +
+      "</figcaption>",
+    '<span class="dot"></span>',
+    '<figcaption class="figure-caption photo-date">' +
+      photo.date +
+      "</figcaption>",
+    "</div>",
+    "</figure>",
+    "</div>",
+  ];
+  return $(component.join(""));
+}
+
+const allComponents = []
+
+const monthToNum = {
+  Jan: 0,
+  Feb: 1,
+  Mar: 2,
+  Apr: 3,
+  May: 4,
+  Jun: 5,
+  Jul: 6,
+  Aug: 7,
+  Sept: 8,
+  Oct: 9,
+  Nov: 10,
+  Dec: 11,
+};
+
+for (const photo of photosData) {
+  // add epoch time to each photo
+  const date = photo.date.split(" ");
+  const year = date[2];
+  const month = monthToNum[date[0]];
+  const day = date[1].slice(0, -2);
+  const epochTime = new Date(year, month, day).getTime() / 1000;
+  photo.epoch = epochTime;
+  // create photo component
+  const component = createPhoto(photo);
+  photo.component = component;
```

Comment:
I think this looks great, no mapping with ids.
please add `@property {Component} component` to the Photo typedef